bot/message_processor.go

```diff
@@ -2,116 +2,139 @@ package bot
-	for _, masterUser := range bot.masterList {
-		if masterUser == user {
-			ismasterUser = true
-		}
+	botID := bot.Slack.GetSelfID()
+	// Handle Violators
+	userID := bot.violatorUserMap.Get(user)
+	if userID != "" {
+		bot.violatorMessage(message, user)
 	}
 
-	switch {
-	// Master user commands
-	case ismasterUser && strings.HasPrefix(message.Text, fmt.Sprintf("<@%s>", bot.Slack.ID)):
-		{
-			bot.masterMessages(message)
+	botCalled := strings.HasPrefix(
+		message.Text,
+		fmt.Sprintf("<@%s>", botID),
+	)
+	if botCalled { // Messages made directly to bot
+		switch {
+		case bot.isMasterUser(user):
+			{
+				bot.masterMessages(message)
+			}
+		default:
+			{
+				bot.niceMessage(message, user)
+			}
 		}
-	// If the user is an offending user message them, but remove them off the list
-	case isViolator:
-		{
-			bot.violatorMessage(message, user)
-		}
-		// If this is a message directed at the bot respond in a nice way
-	case strings.HasPrefix(message.Text, fmt.Sprintf("<@%s>", bot.Slack.ID)):
-		{
-			bot.niceMessage(message, user)
+	} else {
+		switch {
+		// Messages that contain the word tick
+		case strings.Contains(message.Text, " tick "):
+			{
+				bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(
+					"tock", message.Channel,
+				))
+			}
+		// Messages that references the bot will be send out 30% of the time. See: Foucault, Discipline 201
+		case strings.Contains(message.Text, fmt.Sprintf("<@%s>", botID)):
+			{
+				var returnMessage string
+				randomInt := rand.Intn(100)
+				if randomInt >= 70 {
+					returnMessage = bot.MessageRepo.Nice.GenerateMessage(user)
+				} else if randomInt <= 3 {
+					returnMessage = "'The [tockers] must never know whether [they are] looked at at any one moment;"
+					returnMessage += " but [they] must be sure that [they] may always be so' - Foucault, Discipline 201"
+				}
+				bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(
+					returnMessage,
+					message.Channel,
+				))
+			}
 		}
 	}
 }
 
+// violatorMessage has the message for a late user
+func (bot *Bot) violatorMessage(message *slack.MessageEvent, user string) {
+	var returnMessage string
+	// Check if user is still late
+	if bot.isLateUser(user) {
+		returnMessage = bot.MessageRepo.Angry.GenerateMessage(user)
+	} else {
+		returnMessage = fmt.Sprintf(
+			"<@%s>, I was about to yell at you, but then I realized you actually filled out your timesheet. Thanks! ^_^",
+			user,
+		)
+	}
+	bot.violatorUserMap.Delete(user)
+	bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(returnMessage, message.Channel))
+}
+
 // masterMessages contains the commands for admins
-func (bot *Bot) masterMessages(message slackPackage.Message) {
+func (bot *Bot) masterMessages(message *slack.MessageEvent) {
+	var returnMessage string
+	botID := bot.Slack.GetSelfID()
 	switch {
 	case strings.Contains(message.Text, "slap users"):
 		{
 			go bot.SlapLateUsers()
-			message.Text = "Slapping Users!"
+			returnMessage = "Slapping Users!"
```

Comment:
for now, i'm fine with leaving these hard coded here. when you start making tests you'll want to make these const variables then you can easily just check for equality by referencing the var name.